libs/application-generic/src/services/analytic-logs/workflow-run/workflow-run.repository.ts

```diff
@@ -173,50 +180,48 @@ export class WorkflowRunRepository extends LogRepository<typeof workflowRunSchem
         return;
       }
 
-      const existingRuns = await this.find({
-        where: [
-          { workflow_run_id: { operator: '=', value: workflowRunId } },
-          { organization_id: { operator: '=', value: context.organizationId } },
-          { environment_id: { operator: '=', value: context.environmentId } },
-        ],
-        limit: 1,
-        useFinal: true,
+      const notification = await this.notificationRepository.findOne({
+        _id: workflowRunId,
+        _organizationId: context.organizationId,
+        _environmentId: context.environmentId,
       });
 
-      if (existingRuns.data.length === 0) {
-        this.logger.warn(`Workflow run ${workflowRunId} not found for status update`);
+      if (!notification) {
+        this.logger.warn(
+          {
+            workflowRunId,
+            organizationId: context.organizationId,
+            environmentId: context.environmentId,
+          },
+          'Notification not found for workflow run status update'
+        );
+        return;
+      }
+
+      const template = await this.notificationTemplateRepository.findById(
+        notification._templateId,
+        context.environmentId
+      );
```

Comment:
Is it possible to have more stricter mongo selects to return the values we need instead of the whole template?